ddtrace/contrib/bottle/trace.py

```diff
@@ -58,7 +62,13 @@ def wrapped(*args, **kwargs):
                     code = 500
                     raise
                 finally:
-                    response_code = code or response.status_code
+                    if code or isinstance(result, HTTPResponse):
```

Comment:
i'd change this to 

```
if code
    response_code = code
elif isinstance(...)
    response_code = result.status_code
else: 
    ...
```

much clearer. its very weird to combine the Ors all over the place. easier to read.